futures/tests/auto_traits.rs

```diff
@@ -1268,17 +1268,29 @@ pub mod stream {
     assert_impl!(ForEachConcurrent<(), PhantomPinned, PhantomPinned>: Unpin);
     assert_not_impl!(ForEachConcurrent<PhantomPinned, (), ()>: Unpin);
 
-    assert_impl!(Forward<SendTryStream<()>, ()>: Send);
-    assert_not_impl!(Forward<SendTryStream, ()>: Send);
-    assert_not_impl!(Forward<SendTryStream<()>, *const ()>: Send);
-    assert_not_impl!(Forward<LocalTryStream, ()>: Send);
-    assert_impl!(Forward<SyncTryStream<()>, ()>: Sync);
-    assert_not_impl!(Forward<SyncTryStream, ()>: Sync);
-    assert_not_impl!(Forward<SyncTryStream<()>, *const ()>: Sync);
-    assert_not_impl!(Forward<LocalTryStream, ()>: Sync);
-    assert_impl!(Forward<UnpinTryStream, ()>: Unpin);
-    assert_not_impl!(Forward<UnpinTryStream, PhantomPinned>: Unpin);
-    assert_not_impl!(Forward<PinnedTryStream, ()>: Unpin);
+    assert_impl!(Forward<SendStream<()>, ()>: Send);
+    assert_not_impl!(Forward<SendStream, ()>: Send);
+    assert_not_impl!(Forward<SendStream<()>, *const ()>: Send);
+    assert_not_impl!(Forward<LocalStream, ()>: Send);
+    assert_impl!(Forward<SyncStream<()>, ()>: Sync);
+    assert_not_impl!(Forward<SyncStream, ()>: Sync);
+    assert_not_impl!(Forward<SyncStream<()>, *const ()>: Sync);
+    assert_not_impl!(Forward<LocalStream, ()>: Sync);
+    assert_impl!(Forward<UnpinStream, ()>: Unpin);
+    assert_not_impl!(Forward<UnpinStream, PhantomPinned>: Unpin);
+    assert_not_impl!(Forward<PinnedStream, ()>: Unpin);
+
+    assert_impl!(TryForward<SendTryStream<()>, ()>: Send);
```

Comment:
I'm not sure why this is failing and the previous assertion for `Forward` was not?
```
error[E0277]: `*const ()` cannot be sent between threads safely
    --> futures/tests/auto_traits.rs:1283:5
     |
1283 |     assert_impl!(TryForward<SendTryStream<()>, ()>: Send);
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |     |
     |     `*const ()` cannot be sent between threads safely
     |     required by this bound in `stream::_::{closure#0}::assert_impl_all`
     |
     = help: within `futures::stream::TryForward<Pin<Box<(dyn futures::Stream<Item = Result<(), *const ()>> + std::marker::Send + 'static)>>, ()>`, the trait `std::marker::Send` is not implemented for `*const ()`
     = note: required because it appears within the type `Result<(), *const ()>`
     = note: required because it appears within the type `Option<Result<(), *const ()>>`
     = note: required because it appears within the type `futures::stream::try_stream::try_forward::TryForward<Pin<Box<(dyn futures::Stream<Item = Result<(), *const ()>> + std::marker::Send + 'static)>>, (), Result<(), *const ()>>`
     = note: required because it appears within the type `futures::stream::TryForward<Pin<Box<(dyn futures::Stream<Item = Result<(), *const ()>> + std::marker::Send + 'static)>>, ()>`
     = note: this error originates in the macro `assert_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `*const ()` cannot be shared between threads safely
    --> futures/tests/auto_traits.rs:1287:5
     |
1287 |     assert_impl!(TryForward<SyncTryStream<()>, ()>: Sync);
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |     |
     |     `*const ()` cannot be shared between threads safely
     |     required by this bound in `stream::_::{closure#0}::assert_impl_all`
     |
     = help: within `futures::stream::TryForward<Pin<Box<(dyn futures::Stream<Item = Result<(), *const ()>> + Sync + 'static)>>, ()>`, the trait `Sync` is not implemented for `*const ()`
     = note: required because it appears within the type `Result<(), *const ()>`
     = note: required because it appears within the type `Option<Result<(), *const ()>>`
     = note: required because it appears within the type `futures::stream::try_stream::try_forward::TryForward<Pin<Box<(dyn futures::Stream<Item = Result<(), *const ()>> + Sync + 'static)>>, (), Result<(), *const ()>>`
     = note: required because it appears within the type `futures::stream::TryForward<Pin<Box<(dyn futures::Stream<Item = Result<(), *const ()>> + Sync + 'static)>>, ()>`
     = note: this error originates in the macro `assert_impl` (in Nightly builds, run with -Z macro-backtrace for more info)
```